README.md

```diff
@@ -51,43 +73,51 @@ The plugin will emit an error if it detects that it it missed a `loadstart` even
+First you call `player.ads()` to initialize the plugin. Afterwards, the flow of interaction
+between your ad integration and contrib-ads looks like this:
+
+* Player triggers `play` (EVENT) -- This media event is triggered when there is a request to play your player.
+videojs-contrib-ads responds by preventing content playback and showing a loading spinner.
+* Integration triggers `adsready` (EVENT) -- Your integration should trigger this event on the player to indicate that
+it is initialized. This can happen before or after the `play` event.
+* Contrib Ads triggers `readyforpreroll` (EVENT) -- This event is fired after both `play` and `adsready` have ocurred.
+This signals that the integration may begin an ad break by calling `startLinearAdMode`.
+* Integration calls `player.ads.startLinearAdMode()` (METHOD) -- This begins an ad break. During this time, your integration
+plays ads. videojs-contrib-ads does not handle actual ad playback.
+* Integration triggers `ads-ad-started` (EVENT) - Trigger this when each individual ad begins. This removes the loading spinner, which otherwise stays up during the ad break. It's possible for an ad break
+to end without an ad starting, in which case the spinner stays up the whole time.
+* Integration calls `player.ads.endLinearAdMode()` (METHOD) -- This ends an ad break. As a result, content will play.
+* Content plays.
+* To play a Midroll ad, start an end an ad break with `player.ads.startLinearAdMode()` and `player.ads.endLinearAdMode()` at any time during content playback.
+* Contrib Ads triggers `contentended` (EVENT) -- This event means that it's time to play a postroll ad.
+* To play a Postroll ad, start an end an ad break with `player.ads.startLinearAdMode()` and `player.ads.endLinearAdMode()`.
+* Contrib Ads triggers `ended` (EVENT) -- This standard media event happens when all ads and content have completed. After this, no additional ads are expected, even if the user seeks backwards.
+
+This is the basic flow for a simple use case, but there are other things the integration can do:
+
+* `skipLinearAdMode` (METHOD) -- At a time when `startLinearAdMode` is expected, calling `skipLinearAdMode` will immediately resume content playback instead.
+* `nopreroll` (EVENT) -- You can trigger this event even before `readyforpreroll` to indicate that no preroll will play. The ad plugin will not check for prerolls and will instead immediately begin content playback after the `play` event (or immidiately, if playback was already requested).
+* `nopostroll` (EVENT) -- Similar to `nopreroll`, you can trigger this event even before `contentended` to indicate that no postroll will play.  The ad plugin will not wait for a postroll to play and will instead immediately trigger the `ended` event.
+* `adserror` (EVENT) -- This event skips prerolls when seen before a preroll ad break. It skips postrolls if called after contentended and before a postroll ad break. It ends linear ad mode if seen during an ad break.
```

Comment:
We should say when the integration should trigger this event

Reply:
I don't think there is a clear answer to that, it depends on if you want the extra behavior or not. I think we need to change how adserror works. For now, just want to make it clear what it does.

Reply:
To clarify, for prerolls and postrolls the break is skipped if it is triggered before ad mode or before the ad break is started? I think that's not clear for prerolls

Reply:
It says before the ad break, which is exactly accurate.